### PR TITLE
Load Item Worths from the DB

### PR DESCRIPTION
**Pull request by Arrow768**
Adds the option to load the Item Worths from the DB.
Falls back to the Hard-Coded list if the DB is unavailable.

Has a option to seed the DB from the hard coded list.
There is a option to choose if you want to overwrite the existing entries in the table or not.
Seeding the DB needs to be done by calling seed_database() on the Item Worth Subsystem

SQL Migration number is 11, because there is another pull that wants to modify the DB.

TODO: Load the Modifiers from the DB

### CODE REVIEW

File: code/controllers/subsystems/item_worths.dm [hunk 1]
```diff
@@ -1,17 +1,77 @@
-//Negative values indicate that instances of these types should use the Value proc
-//Mainly used so that stuff inside them can also add to their value, and other things like material,
-//stuff like that.
+/var/global/datum/controller/subsystem/item_worths/SSItemWorths
```
Comment (Lohikar) [inline code/controllers/subsystems/item_worths.dm]: This should be `SSitem_worths`.

File: code/controllers/subsystems/item_worths.dm [hunk 2]
```diff
@@ -1,17 +1,77 @@
...
+	name = "Item Worths"
+	flags = SS_NO_FIRE
+	init_order = SS_INIT_MISC_FIRST
+
+	var/list/worths = list()
+
+/datum/controller/subsystem/item_worths/New()
+	NEW_SS_GLOBAL(SSItemWorths)
+
+/datum/controller/subsystem/item_worths/Initialize(timeofday)
```
Comment (Lohikar) [inline code/controllers/subsystems/item_worths.dm]: Must call parent at the end of this.

File: code/controllers/subsystems/item_worths.dm [hunk 3]
```diff
@@ -1,17 +1,77 @@
...
-//Mainly used so that stuff inside them can also add to their value, and other things like material,
-//stuff like that.
+/var/global/datum/controller/subsystem/item_worths/SSItemWorths
 
-var/list/worths = list(
-//ROBOT ASSEMBLIES,
+/datum/controller/subsystem/item_worths
+	name = "Item Worths"
+	flags = SS_NO_FIRE
+	init_order = SS_INIT_MISC_FIRST
```
Comment (Lohikar) [inline code/controllers/subsystems/item_worths.dm]: This need to be this early?

File: code/controllers/subsystems/item_worths.dm [hunk 4]
```diff
@@ -1,17 +1,77 @@
...
+			var/path = text2path(text_path)
+			var/value = item_worth_query.item[2]
+			if(path)
+				worths[path] = value
+			else
+				log_debug("SSItemWorths: Loading Error: [text_path] is not a valid path")
+
+/datum/controller/subsystem/item_worths/proc/seed_database(overwrite = 0)
+	if(config.item_worths_load_from == "sql")
+		log_debug("SSItemWorths: SEED ERROR: You need to load from code, if you wish to seed the SQL DB.")
```
Comment (Lohikar) [inline code/controllers/subsystems/item_worths.dm]: Shouldn't this be WARNING or something other than debug?

File: code/controllers/subsystems/item_worths.dm [hunk 5]
```diff
@@ -1,17 +1,77 @@
...
+		return 1
+	if(!establish_db_connection(dbcon))
+		log_debug("SSItemWorths: SQL ERROR: Failed to connect ! - Unable to seed DB.")
```
Comment (Lohikar) [inline code/controllers/subsystems/item_worths.dm]: Same here.

File: code/controllers/subsystems/item_worths.dm (hunk 3 above)
Comment (Arrow768) [reply code/controllers/subsystems/item_worths.dm]: Yes, because other systems will require the item values (for example vendors)

**Pull request closed without merging**